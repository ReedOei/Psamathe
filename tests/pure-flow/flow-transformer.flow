transformer newAccount(a : any nat) -> res : ! list ! { name : any nat, balance : any nat } {
    [ ! { name : any nat, balance : any nat } ;
        {
            name : any nat |-> a,
            balance : any nat |-> 0
        }
    ] --> res
}

;

[ any nat ; 0,1,2 ] --> newAccount() --> var accounts : list ! { name : any nat, balance : any nat }

