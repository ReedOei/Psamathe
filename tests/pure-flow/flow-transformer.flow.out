<generatedTop>
  <common>
    <typeEnv>
      .Map
    </typeEnv>
    <types>
      .Map
    </types>
    <functions>
      newAccount |-> transformer ( res : ! table ( .Vars ) ! record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , a : any nat , .VarDefs , [ ! record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } ; record ( .Vars ) { name : any nat |-> a , balance : any nat |-> 0 , .RecordMembers } , .Vars ] --> res  .Stmts )
    </functions>
  </common>
  <k>
    .
  </k>
  <lookup>
    accounts |-> selected ( loc ( 3 ) , loc ( 3 ) )
  </lookup>
  <storage>
    loc ( 0 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 1 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 2 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 3 ) |-> resource ( table ( .Vars ) ! record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , ListItem ( loc ( 9 ) )
    ListItem ( loc ( 15 ) )
    ListItem ( loc ( 21 ) ) )
    loc ( 4 ) |-> resource ( table ( .Vars ) ! record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , ListItem ( loc ( 9 ) ) )
    loc ( 5 ) |-> resource ( record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , .List )
    loc ( 6 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 7 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 8 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 9 ) |-> resource ( record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , ListItem ( balance |-> loc ( 8 )
    name |-> loc ( 6 ) ) )
    loc ( 10 ) |-> resource ( table ( .Vars ) ! record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , ListItem ( loc ( 15 ) ) )
    loc ( 11 ) |-> resource ( record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , .List )
    loc ( 12 ) |-> resource ( nat , ListItem ( 1 ) )
    loc ( 13 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 14 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 15 ) |-> resource ( record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , ListItem ( balance |-> loc ( 14 )
    name |-> loc ( 12 ) ) )
    loc ( 16 ) |-> resource ( table ( .Vars ) ! record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , ListItem ( loc ( 21 ) ) )
    loc ( 17 ) |-> resource ( record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , .List )
    loc ( 18 ) |-> resource ( nat , ListItem ( 2 ) )
    loc ( 19 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 20 ) |-> resource ( nat , ListItem ( 0 ) )
    loc ( 21 ) |-> resource ( record ( .Vars ) { name : any nat , balance : any nat , .VarDefs } , ListItem ( balance |-> loc ( 20 )
    name |-> loc ( 18 ) ) )
  </storage>
  <catchStack>
    .List
  </catchStack>
  <createdVals>
    .Map
  </createdVals>
</generatedTop>
